src/java/org/apache/ivy/plugins/conflict/LatestCompatibleConflictManager.java

```diff
@@ -85,14 +85,12 @@ public LatestCompatibleConflictManager(String name, LatestStrategy strategy) {
         if (versionMatcher.isDynamic(mrid)) {
             while (iter.hasNext()) {
                 IvyNode other = iter.next();
-                if (versionMatcher.isDynamic(other.getResolvedId())) {
+                if (versionMatcher.isDynamic(other.getResolvedId())
+                        || !versionMatcher.accept(mrid, other.getResolvedId())
+                        && !handleIncompatibleConflict(parent, conflicts, node, other)) {
```

Comment:
[It still does](http://introcs.cs.princeton.edu/java/11precedence/). There's a difference between `&&` and `||` vs `&` and `|`, see "short circuiting".